packages/sdk/src/libs/clients/registry/registry.ts

```diff
@@ -0,0 +1,59 @@
+import assert from "assert";
+import { Registry__factory } from "@uma/core/contract-types/ethers";
+import Registry from "@uma/core/build/contracts/Registry.json";
+import type { SignerOrProvider } from "../../index.d";
+import { EventFragment, Result } from "@ethersproject/abi";
+import { BlockTag } from "@ethersproject/abstract-provider";
+import { EventFilter, Event } from "ethers";
+
+export interface TypedEvent<EventArgs extends Result> extends Event {
```

Comment:
This is really annoying. They really don't have great support for events.

Could we build something like this to make an auto-retrieval of the type? (reference: https://stackoverflow.com/a/62733441). Note: this is untested.

```ts

// We could put all of this in some utils file so it could be 
import type { TypedEventFilter } from "@uma/core/contract-types/ethers/commons";
import { Registry } from "@uma/core/contract-types/ethers/Registry";
// We may be able to make this slightly easier to use by allowing the caller to pass the ContractType and the
// name of the Event, but I'm not sure exactly how that syntax would work.
type GetEventType<EventFilterFunction> = ReturnType<EventFilter> extends TypedEventFilter<infer T> ? T : never; 


....

// There may be cleaner ways to access the properties on Registry.
type NewContractRegistered = GetEventType<Registry["filters"]["NewContractRegistered"]>;
```

Reply:
will play around with this and see if there is a way thanks

Reply:
the main blocker here is that i cant figure out a way to get return type from a generic, theres some discussion here, but its hard to wrap my head around: https://stackoverflow.com/questions/50321419/typescript-returntype-of-generic-function

Reply:
actually i think i figured it out: `type GetEventType<F> = ReturnType<F extends Callable ? F : never>extends TypedEventFilter<infer T, infer S> ? TypedEvent<T & S & Result> : never;`

this is compiling, will give it a go

Reply:
it compiles and runs, but doesnt properly typecheck things. I tested by accessing some props which should have been invalid, but typescript did not throw an error.